lib/src/components/board/board.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:viam_sdk/src/resource/base.dart';
+
+import '../../gen/common/v1/common.pb.dart' as common;
+import '../../robot/client.dart';
+
+class BoardStatus {
+  final Map<String, int> analogs;
+  final Map<String, int> digitalInterrupts;
+
+  BoardStatus(this.analogs, this.digitalInterrupts);
+
+  factory BoardStatus.fromProto(common.BoardStatus pbBoardStatus) {
+    BoardStatus boardStatus = BoardStatus(Map<String, int>(), Map<String, int>());
+    pbBoardStatus.analogs.forEach((key, value) => boardStatus.analogs[key] = value.value);
+    pbBoardStatus.digitalInterrupts.forEach((key, value) => boardStatus.digitalInterrupts[key] = value.value.toInt());
```

Comment:
`boardStatus.digitalInterrupts[key] = value.value as int);`

was giving me problems here, throwing errors around casting for some reason. while digging around i discovered that `Int64` has a [toInt() method](https://pub.dev/documentation/fixnum/latest/fixnum/Int64/toInt.html) - this may be more preferable to use rather than casting ourselves. Something to think about, I will put up a follow up converting the code base to use this and that discussion can be continued there if need be.

Reply:
I agree, it was confusing for me. I'm having some more ideas around this I want to dig into to understand 👀 - wondering if the `as int` was trying to caste more of the statement than what I wanted, but that is just a hunch.

Reply:
☝️ that is what it was. refer to the current version, but wrapping the Int64 and `as int` in `()` made it work as expected. I wasn't wrapping it before and it was trying to convert the whole statement to int.